src/main/java/org/pesmypetcare/webservice/dao/FreqTrainingDaoImpl.java

```diff
@@ -0,0 +1,162 @@
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        if (!mealDoc.exists()) {
+            throw new DatabaseAccessException(INVALID_FREQTRAINING_EXC, FREQTRAINING_DOES_NOT_EXIST_EXC);
+        }
+        return mealDoc.toObject(FreqTrainingEntity.class);
+    }
+
+    @Override
+    public List<Map<String, Object>> getAllFreqTraining(String owner, String petName) throws DatabaseAccessException {
+        CollectionReference freqTrainingsRef = getFreqTrainingsRef(owner, petName);
+        List<Map<String, Object>> externalList = new ArrayList<>();
+        try {
+            getAllFreqTrainingsOfAPetFromDatabase(freqTrainingsRef, externalList);
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        return externalList;
+    }
+
+    @Override
+    public List<Map<String, Object>> getAllFreqTrainingsBetween(String owner, String petName, String initialDate,
+                                                          String finalDate) throws DatabaseAccessException {
+        CollectionReference freqTrainingsRef = getFreqTrainingsRef(owner, petName);
+        List<Map<String, Object>> externalList = new ArrayList<>();
+        try {
+            getFreqTrainingsBetweenDatesFromDatabase(initialDate, finalDate, freqTrainingsRef, externalList);
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        return externalList;
+    }
+
+    @Override
+    public void updateFreqTraining(String owner, String petName, String petDate, Object value) {
+        CollectionReference freqTrainingsRef = getFreqTrainingsRef(owner, petName);
+        DocumentReference freqTrainingRef = freqTrainingsRef.document(petDate);
+        freqTrainingRef.update("freqTrainingValue", value);
+    }
+
+    /**
+     * Return the freqTraining collection of one pet.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @return Return the freqTraining collection of one pet
+     */
+    public CollectionReference getFreqTrainingsRef(String owner, String petName) {
+        return db.collection("users").document(owner).collection("pets").document(petName)
+            .collection("freqTrainings");
+    }
+
+    /**
+     * Gets all the freqTrainings of the collection and puts them in the externalList.
+     * @param freqTrainingsRef Reference to the collection of freqTrainings
+     * @param externalList list that will contain all the freqTrainings
+     * @throws InterruptedException Exception thrown by the DB if the operation is interrupted
+     * @throws ExecutionException Exception thrown by the DB if there's an execution problem
+     */
+    private void getAllFreqTrainingsOfAPetFromDatabase(CollectionReference freqTrainingsRef, List<Map<String, Object>> externalList)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 132).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5400494)